unlock-app/src/components/interface/keychain/KeyDetails.tsx

```diff
@@ -1,30 +1,45 @@
 import React, { useContext } from 'react'
-import { ApolloProvider, useQuery } from '@apollo/react-hooks'
 import { AuthenticationContext } from '../../../contexts/AuthenticationContext'
-import keyHolderQuery from '../../../queries/keyHolder'
 import 'cross-fetch/polyfill'
 import { DefaultError } from '../../creator/FatalError'
-import { OwnedKey } from './KeychainTypes'
 import Key from './Key'
 import LoginPrompt from '../LoginPrompt'
 import networks from '@unlock-protocol/networks'
-import ApolloClient from 'apollo-boost'
-import { NetworkConfig } from '@unlock-protocol/types'
-
+import { SubgraphService } from '@unlock-protocol/unlock-js'
+import { useQuery } from '@tanstack/react-query'
+import { ImageBar } from '../locks/Manage/elements/ImageBar'
+import { useConfig } from '~/utils/withConfig'
 interface KeysByNetworkProps {
   account: string
-  network: NetworkConfig
+  network: number
 }
+
 export const KeysByNetwork = ({ account, network }: KeysByNetworkProps) => {
-  const { loading, error, data } = useQuery(keyHolderQuery(), {
-    variables: { address: account },
-  })
+  const { networks } = useConfig()
+  const networkName = networks[network]?.name
+  const getKeys = async () => {
+    const service = new SubgraphService()
 
-  const { name, id } = network
+    return await service.keys(
+      {
+        first: 1000,
+        where: {
+          owner: account,
+        },
+      },
+      {
+        networks: [`${network}`],
```

Comment:
You can add `number[] | string[]` to the subgraph service type so you don't need to do this here.